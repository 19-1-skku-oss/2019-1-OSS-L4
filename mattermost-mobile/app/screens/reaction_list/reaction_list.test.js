// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import SlideUpPanel from 'app/components/slide_up_panel';

import ReactionList from './reaction_list';

jest.mock('react-intl');

describe('ReactionList', () => {
    const baseProps = {
        actions: {
            getMissingProfilesByIds: jest.fn(),
        },
        allUserIds: ['user_id_1', 'user_id_2'],
        navigator: {setOnNavigatorEvent: jest.fn()},
        reactions: {'user_id_1-smile': {emoji_name: 'smile', user_id: 'user_id_1'}, 'user_id_2-+1': {emoji_name: '+1', user_id: 'user_id_2'}},
        theme: Preferences.THEMES.default,
        teammateNameDisplay: 'username',
        userProfiles: [{id: 'user_id_1', username: 'username_1'}, {id: 'user_id_2', username: 'username_2'}],
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <ReactionList {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(SlideUpPanel).exists()).toEqual(true);
    });

    test('should match snapshot, renderReactionRows', () => {
        const wrapper = shallow(
            <ReactionList {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.instance().renderReactionRows()).toMatchSnapshot();
    });

    test('should match state on handleOnSelectReaction', () => {
        const wrapper = shallow(
            <ReactionList {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.setState({selected: 'smile'});
        wrapper.instance().handleOnSelectReaction('+1');
        expect(wrapper.state('selected')).toEqual('+1');
    });
});
