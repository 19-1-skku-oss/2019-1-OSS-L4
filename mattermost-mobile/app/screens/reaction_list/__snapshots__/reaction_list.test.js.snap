// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionList should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ForwardRef(forwardConnectRef)
    header={[Function]}
    headerHeight={37.5}
    initialPosition={0.55}
    onRequestClose={[Function]}
  >
    <View
      style={
        Object {
          "height": 45,
          "justifyContent": "center",
        }
      }
    >
      <ReactionRow
        emojiName="+1"
        navigator={
          Object {
            "setOnNavigatorEvent": [MockFunction] {
              "calls": Array [
                Array [
                  [Function],
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
          }
        }
        teammateNameDisplay="username"
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        user={
          Object {
            "id": "user_id_2",
            "username": "username_2",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.2)",
            "height": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "height": 45,
          "justifyContent": "center",
        }
      }
    >
      <ReactionRow
        emojiName="smile"
        navigator={
          Object {
            "setOnNavigatorEvent": [MockFunction] {
              "calls": Array [
                Array [
                  [Function],
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": undefined,
                },
              ],
            },
          }
        }
        teammateNameDisplay="username"
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        user={
          Object {
            "id": "user_id_1",
            "username": "username_1",
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(61,60,64,0.2)",
            "height": 1,
          }
        }
      />
    </View>
  </ForwardRef(forwardConnectRef)>
</View>
`;

exports[`ReactionList should match snapshot, renderReactionRows 1`] = `
Array [
  <View
    style={
      Object {
        "height": 45,
        "justifyContent": "center",
      }
    }
  >
    <ReactionRow
      emojiName="+1"
      navigator={
        Object {
          "setOnNavigatorEvent": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      teammateNameDisplay="username"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      user={
        Object {
          "id": "user_id_2",
          "username": "username_2",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "height": 1,
        }
      }
    />
  </View>,
  <View
    style={
      Object {
        "height": 45,
        "justifyContent": "center",
      }
    }
  >
    <ReactionRow
      emojiName="smile"
      navigator={
        Object {
          "setOnNavigatorEvent": [MockFunction] {
            "calls": Array [
              Array [
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
        }
      }
      teammateNameDisplay="username"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      user={
        Object {
          "id": "user_id_1",
          "username": "username_1",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "height": 1,
        }
      }
    />
  </View>,
]
`;
